pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -55,268 +136,511 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	pred := predicate.Funcs{
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			newObj := e.ObjectNew.(*kubevirtv1alpha1.NodeMaintenance)
-			return !newObj.DeletionTimestamp.IsZero()
-		},
-	}
+	return c.Watch(&source.Kind{Type: &corev1.Node{}}, &handler.EnqueueRequestForObject{})
+}
+
+// Reconcile monitors Nodes and creates the MachineRemediation object when the node has reboot annotation.
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 
-	// Create a source for watching noe maintenance events.
-	src := &source.Kind{Type: &kubevirtv1alpha1.NodeMaintenance{}}
+	r.reqLogger = log.WithFields(log.Fields{"Request.Namespace": request.Namespace, "Request.Name": request.Name})
+	r.reqLogger.Debug("Reconciling node", request.Name)
+
+	// Get node from request
+	node := &corev1.Node{}
+
+	if err := r.client.Get(context.TODO(), request.NamespacedName, node); err != nil {
+		if errors.IsNotFound(err) {
+			r.reqLogger.Info("node not found")
+			return reconcile.Result{}, nil
+		}
+		r.reqLogger.Errorf("node not found. name %s error %v", request.Name, err)
+		return reconcile.Result{}, err
+	}
 
-	// Watch for changes to primary resource NodeMaintenance
-	err = c.Watch(src, &handler.EnqueueRequestForObject{}, pred)
+	// parse status & spec, create initial status object on first call
+	specAnnoData, statusAnnoData, node, err := r.parseAnnotations(node)
 	if err != nil {
-		return err
+		r.reqLogger.Errorf("invalid value of annotation. name %s error %v", request.Name, err)
```

Comment:
``request.Name`` is logged with every call.
Logging the annotation name here would be worthwhile though.

Reply:
the value of the annotation is included in the error that is passed up to this level and logged with this call.

Reply:
changed it so that the value of the annotation will be in the error passed up by parseAnnotation and that it printed with this call.

Reply:
I suggested logging the annotation _name_, not it's contents